rust/lk/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+//! Rust interfaces to LK
+
+#![no_std]
+
+extern crate alloc;
+
+pub extern crate lk_sys as sys;
+
+use alloc::format;
+use core::ptr::addr_of_mut;
+use core::{
+    alloc::{GlobalAlloc, Layout},
+    ffi::{CStr, c_uint, c_void},
+    panic::PanicInfo,
+};
+
+pub mod cbuf;
+pub mod init;
+pub mod lkonce;
+pub mod log;
+pub mod macros;
+
+/// Initialize the rust side of the code.
+///
+/// Due to linker optimization, without a direct call into this code, the linker will not bring in
+/// anything.
+pub fn init() {
```

Comment:
Though it's not doing much, I think i mentioned it in the platform code, we should probably call this as soon as possible, so that any rust code that comes online later can just assume any background rust setup has already happened. OTOH, this fputs may be assuming there's already some sort of log path available? May need to init silently.

Reply:
Realistically, this probably shouldn't even be called 'init', since it isn't actually doing anything, just making sure the code is linked in. The init can happen through the normal lk init mechanism.